### PR TITLE
[YLD-106][Remove Liquidity Modal]

### PR DESCRIPTION
**Pull request by desamtralized**

### CODE REVIEW

File: app/app/explore/components/LiquidityPoolCardMobile.tsx
Comment (desamtralized) [inline app/app/explore/components/LiquidityPoolCardMobile.tsx:1]: pnpm format

File: app/app/explore/pools/[asset]/PoolDetail.tsx [hunk 1]
```diff
@@ -223,7 +161,43 @@ export default function PoolDetail({ pool, runePriceUSD }: PoolDetailProps) {
...
+      const runeAmount = formatNumber(positionDetails.runeAdded, DECIMALS);
+
+      return (
+        <div key={position.liquidityUnits} className="mb-4">
+          <div className="flex-col items-center text-base font-medium text-neutral-900 mb-1">
+            <div>
+              {assetAmount} {assetSymbol} + {runeAmount} RUNE
+            </div>
+          </div>
+          <button
```
Comment (MattNguyen) [inline app/app/explore/pools/[asset]/PoolDetail.tsx:182]: This should be part of the generalized button component

File: app/app/explore/components/RemoveLiquidityModal.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                  : "bg-white text-secondaryBtn",
+              )}
+              disabled={isSubmitting}
+            >
+              {percent === 100 ? "MAX" : `${percent}%`}
+            </button>
+          ))}
+        </div>
+
+        <button
```
Comment (MattNguyen) [inline app/app/explore/components/RemoveLiquidityModal.tsx:164]: This should be part a generalized button component

File: app/app/utils.tsx [hunk 3]
```diff
@@ -540,3 +540,20 @@ export const calculateGain = async (
...
   }
 };
+
+export interface PositionDetails {
+  assetAdded: number;
+  runeAdded: number;
+}
+
+export const getPositionDetails = (
+  pool: PoolDetail,
```
Comment (MattNguyen) [inline app/app/utils.tsx:550]: Why do you need pool details?

File: app/app/utils.tsx [hunk 4]
```diff
@@ -540,3 +540,20 @@ export const calculateGain = async (
...
+  position: MemberPool,
+): PositionDetails => {
+  const assetAdded = parseFloat(position.assetAdded) / DECIMALS;
+  const runeAdded = parseFloat(position.runeAdded) / DECIMALS;
+  return {
+    assetAdded,
+    runeAdded,
+  } as PositionDetails;
```
Comment (MattNguyen) [inline app/app/utils.tsx:558]: Do you need type coercion here?

File: app/app/explore/components/RemoveLiquidityModal.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    setPercentage(percent);
+    const amount = (positionAssetAmount * percent) / 100;
+    setAssetAmount(amount.toString());
+  };
+
+  const handleValueChange = (values: NumberFormatValues) => {
+    setAssetAmount(values.value);
+    const amount = parseFloat(values.value);
+    const percentage = (amount / positionAssetAmount) * 100;
+    setPercentage(percentage);
```
Comment (MattNguyen) [inline app/app/explore/components/RemoveLiquidityModal.tsx:58]: why do we need to persist percentage in useState?

**Comment by MattNguyen** [comment]
Where does the balance come from in the remove liquidity modal? The amount does not appear to equal to deposit+game from the main position details screen.

image

image

**Pull request merged**